drivers/sensor/vl53l0x/vl53l0x.c

```diff
@@ -280,9 +284,20 @@ static int vl53l0x_init(const struct device *dev)
 	return 0;
 }
 
-
-static struct vl53l0x_data vl53l0x_driver;
-
-DEVICE_DT_INST_DEFINE(0, vl53l0x_init, NULL, &vl53l0x_driver,
-		      NULL, POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY,
-		      &vl53l0x_api_funcs);
+#define VL53L0X_INIT(inst)						 \
+	static struct vl53l0x_config vl53l0x_##inst##_config = {	 \
+		.i2c = DEVICE_DT_GET(DT_INST_BUS(inst)),		 \
+		.dt_addr = DT_INST_REG_ADDR(inst),			 \
```

Comment:
You can use i2c_dt_spec. which incorporates i2c bus device and addr, and use I2C_DT_SPEC_INST_GET() to get them. So in .h you can define 

i2c_dt_spec i2c;

And here:
.i2c = I2C_DT_SPEC_INST_GET(inst);

Reply:
why did you mark it as resolved?

Reply:
Good catch, thank you @avisconti 
I messed up while cherry-picking changes back into the PR commits

Reply:
Yes, but not sure my comments is clear. Is it?

Reply:
Yes it is, see https://github.com/zephyrproject-rtos/zephyr/pull/38315/commits/b26d4f83ec8766a0532c0e6df69672e566f226c0#diff-d58092bb6d79c1da0579e21fb5d1eed6322aa15905562ba99c8e7d0927be914bL43-L44 and https://github.com/zephyrproject-rtos/zephyr/pull/38315/commits/b26d4f83ec8766a0532c0e6df69672e566f226c0#diff-d58092bb6d79c1da0579e21fb5d1eed6322aa15905562ba99c8e7d0927be914bL289-L290

Is that what you expected ?

Reply:
The code change is correct, but it's in the wrong commit. If you review the commits one by one, you can see that this first gets added as `.i2c = DEVICE_DT_GET(DT_INST_BUS(inst))` in commit `drivers: sensor: vl53l0x: add driver support for multiple sensors`. Then it gets corrected to `.i2c = I2C_DT_SPEC_INST_GET(inst),` in commit `drivers: sensor: vl53l0x: add support to reprogram I2C address`.

Reply:
OK, I see. I was looking at  "drivers: sensor: vl53l0x: add driver support for multiple sensors" commit only, where I think the change should be done. Can you move it in this commit? I think it behaves more to this one. Thanks and sorry for confusion.